docs/flows/flows.puml

```diff
@@ -272,20 +271,136 @@ bitdaoadmin <-- insurancecontract: execute proposal
 
 newpage
 
-autonumber "Upgrades:[00]"
+autonumber "UpgradeGovernance:[00]"
 
-== Upgrades ==
+== Upgrade Governance ==
 
-bitdaoadmin -> governanceproxy: deploy proxy and set initial implementation
-bitdaoadmin -> treasuryproxy: deploy proxy and set initial implementation
-bitdaoadmin -> executorproxy: deploy proxy and set initial implementation
+bitdaoadmin -> executor: deploy timelock executor
+bitdaoadmin -> bitdaogovernance: deploy initial governance contract logic
+bitdaoadmin -> proxyadmin: deploy admin for governance proxy
+bitdaoadmin -> proxy: deploy governance proxy and set logic and admin
+bitdaoadmin -> proxyadmin: set executor as admin
 
 
 alice -> bitdaogovernance: deploy new bitdaogovernance implementation
+
+alice -> forum: create forum upgrade proposal with new bitdaogovernance
+alice <- community: positive/negative feedback
+
+community -> community: delegate role (developer/all) based votes
+
+alice -> bitdaogovernance: submit upgrade proposal
+community -> bitdaogovernance: roles voting
+
+bitdaoadmin -> bitdaogovernance: get roles voting result
+bitdaoadmin <-- bitdaogovernance: pass/fail vote result
+
+bitdaoadmin -> bitdaogovernance: start execution
+executor <-- bitdaogovernance: execute proposal
+executor -> proxyadmin: call governance proxy upgrade to new bitdaogovernance implementation
```

Comment:
thought: the dependency between governance and other proxied contracts hasn't been indicated. At some point, we'll need to identify the dependency order in which governance, treasury, or other contracts should be updated. 

suggestion: if there is no inter-dependence between contracts, perhaps a note saying so would be helpful

Reply:
I can indicate it with separate call from bitdaoadmin (or we can call this section init/upgrade process) to set governance or other contract as dependency for all those contracts. Executor is just a pair contract for governance to make all the executions.